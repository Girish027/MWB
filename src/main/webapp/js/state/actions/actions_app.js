
import _ from 'lodash';
import getUrl, { pathKey } from 'utils/apiUrls';
import { fetchUtil } from 'utils/fetchUtils';
import goToRoute from 'utils/routeHelpers';
import Constants from 'constants/Constants';
import AmplitudeConstants from 'constants/AmplitudeConstants';
import { logAmplitudeEvent } from 'utils/amplitudeUtils';
import axios from 'axios';
import api from 'utils/api';
import { setDefaultsForAPI } from 'utils/api';
import * as types from './types';

// Note to developers:
// Both Warning and Good Messages are shown for the period of notificationInterval.
// Bad, Persistant - till user closes the notification. This should only be used sparingly.
export const callLogIngestAPI = (message, logLevel) => {
  const promise = new Promise((resolve, reject) => {
    const url = getUrl(pathKey.logIngest, { logLevel });
    return new Promise((resolve, reject) => {
      api.post({
        url,
        headers: {
          'Content-Type': 'text/plain',
        },
        data: message,
      });
    });
  });
};

export const displayBadRequestMessage = (message) => (dispatch) => {
  dispatch({
    type: types.BAD_REQUEST,
    value: message,
  });

  callLogIngestAPI(message, Constants.LOG_LEVEL_ERROR);
};

export const displayWarningRequestMessage = (message, notificationInterval = Constants.NOTIFICATION.interval) => (dispatch) => {
  dispatch({
    type: types.WARNING_REQUEST,
    value: message,
  });

  callLogIngestAPI(message, Constants.LOG_LEVEL_WARNING);

  setTimeout(() => {
    dispatch(stopShowingServerMessage());
  }, notificationInterval);
};

export const displayGoodRequestMessage = (message, notificationInterval = Constants.NOTIFICATION.interval) => (dispatch) => {
  dispatch({
    type: types.GOOD_REQUEST,
    value: message,
  });

  setTimeout(() => {
    dispatch(stopShowingServerMessage());
  }, notificationInterval);
};

export const displayPersistantMessage = message => ({
  type: types.PERSISTENT_REQUEST,
  value: message,
});

export const stopShowingServerMessage = () => ({
  type: types.APP_STOP_SHOWING_SERVER_MESSAGE,
});

export const updateLoginSuccess = csrfToken => ({
  type: types.LOGIN_SUCCESS,
  value: csrfToken,
});

export const updateLoginInfo = (userId, userName, userDetails) => ({
  type: types.LOGIN_INFO,
  value: {
    userId,
    userName,
    userDetails,
  },
});

export const logoutFromOktaSuccess = () => ({
  type: types.LOGOUT_OKTA,
});

export const logoutFromMWBSuccess = () => ({
  type: types.LOGOUT_MWB,
});

export const contextMenuChange = contextMenuState => ({
  type: types.CONTEXT_MENU_CHANGE,
  contextMenuState,
});

export const modalDialogChange = modalDialogState => ({
  type: types.MODAL_DIALOG_CHANGE,
  modalDialogState,
});

const actionFileRemoveRequest = ({ fileId }) => ({
  type: types.FILE_REMOVE_REQUEST,
  fileId,
});

const actionFileRemoveError = ({ fileId, error }) => ({
  type: types.FILE_REMOVE_ERROR,
  fileId,
  error,
});

const actionFileRemoveSuccess = ({ fileId }) => ({
  type: types.FILE_REMOVE_SUCCESS,
  fileId,
});

export const updateUserGroups = data => ({
  type: types.UPDATE_USER_GROUPS,
  data,
});

const onFetchFailureRemoveFile = (error, params) => (dispatch) => {
  const { fileId } = params;
  dispatch(actionFileRemoveError({ fileId, error }));
};

const onFetchSuccessRemoveFile = (data, params) => (dispatch) => {
  const { fileId } = params;
  dispatch(actionFileRemoveSuccess({ fileId }));
};

export const removeFile = ({ fileId }) => (dispatch, getState) => {
  dispatch(actionFileRemoveRequest({ fileId }));

  const state = getState();
  const fetchUrl = getUrl(pathKey.files, { fileId, clientId: state.header.client.id });

  fetchUtil({
    fetchUrl,
    fetchMethod: 'delete',
    onFetchSuccess: onFetchSuccessRemoveFile,
    onFetchFailure: onFetchFailureRemoveFile,
    dispatch,
    getState,
    params: {
      fileId,
    },
  });
};

const onFetchSuccessGetUserGroups = data => (dispatch) => {
  dispatch(updateUserGroups(data));
};

export const getUserGroups = ({ csrfToken }) => (dispatch, getState) => {
  const fetchUrl = getUrl(pathKey.userGroups, { csrfToken });

  fetchUtil({
    fetchUrl,
    fetchMethod: 'get',
    onFetchSuccess: onFetchSuccessGetUserGroups,
    dispatch,
    getState,
  });
};

export const onFetchSuccessGetCSRFToken = data => (dispatch, getState) => {
  const csrfToken = data.value;
  setDefaultsForAPI(csrfToken);
  dispatch(updateLoginSuccess(csrfToken));
  dispatch(getUserGroups({ csrfToken }));
};

export const getCSRFToken = userId => (dispatch, getState) => {
  const fetchUrl = getUrl(pathKey.csrfToken);

  fetchUtil({
    fetchUrl,
    fetchMethod: 'get',
    onFetchSuccess: onFetchSuccessGetCSRFToken,
    dispatch,
    getState,
    params: {
      userId,
    },
  });
};

const logoutFromMWB = () => (dispatch, getState) => {
  dispatch(logoutFromOktaSuccess());
  dispatch(logoutFromMWBSuccess());
};

export const logoutFromOkta = () => (dispatch, getState) => {
  const state = getState();
  const { oAuthLogoutURL } = state.app;
  if (oAuthLogoutURL) {
    fetch(oAuthLogoutURL, {
      method: 'DELETE',
      credentials: 'include',
      headers: {
        'Content-Type': 'application/json',
        Accept: 'application/json',
      },
    })
      .then((response) => {
        window.location.href = Constants.LOGOUT;
        switch (response.status) {
        case 204:
          logoutFromMWB();
          logAmplitudeEvent(AmplitudeConstants.LOGOUT_EVENT, state);
          break;
        case 404:
        default:
          // TODO send log for these scenarios
          break;
        }
      })
      .catch((error) => {
        // TODO send log for these scenarios
      });
  }
};

export const changeRoute = (routeName, data, history) => (dispatch, getState) => {
  const state = getState();
  const routeData = Object.assign({}, {
    client: !_.isNil(state.header.client) ? state.header.client : {
      id: 0,
      name: 'Loading...',
    },
    projectId: state.selectedProjectId,
  }, data);
  goToRoute(routeName, routeData, history);
};
