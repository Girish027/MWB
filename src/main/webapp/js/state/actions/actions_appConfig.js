import 'regenerator-runtime/runtime';

import getUrl, { pathKey } from 'utils/apiUrls';

import { fetchUtil } from 'utils/fetchUtils';
import AmplitudeConstants from 'constants/AmplitudeConstants';
import { initializeAnalytics, logAmplitudeEvent } from 'utils/amplitudeUtils';
import api from 'utils/api';
import * as types from './types';

const appConfigSuccess = data => ({
  type: types.APP_CONFIG_SUCCESS,
  data,
});

const appConfigFailure = data => ({
  type: types.APP_CONFIG_FAILURE,
  data,
});

const appConfigRequest = () => ({
  type: types.APP_CONFIG_REQUEST,
});

export const onAppConfigSuccess = data => (dispatch, getState) => {
  dispatch(appConfigSuccess(data));
  const state = getState();
  initializeAnalytics(state);
  logAmplitudeEvent(AmplitudeConstants.LOGIN_EVENT, state);
};

export const onAppConfigFailure = error => (dispatch) => {
  dispatch(appConfigFailure(error));
};

export const fetchAppConfig = () => (dispatch, getState) => {
  dispatch(appConfigRequest());
  const fetchUrl = getUrl(pathKey.appConfig);

  api.get({
    url: fetchUrl,
    onApiSuccess: onAppConfigSuccess,
    onApiError: onAppConfigFailure,
    dispatch,
    getState,
  });
};
