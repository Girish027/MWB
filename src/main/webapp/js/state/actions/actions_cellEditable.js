import * as types from './types';

export const stateCreate = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_CREATE,
  stateKey,
});

export const stateReset = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_RESET,
  stateKey,
});

export const stateRemove = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_REMOVE,
  stateKey,
});

export const setActiveCell = ({ stateKey, activeRowIndex, activeColumnIndex }) => ({
  type: types.CELL_EDITABLE_SET_ACTIVE_CELL,
  stateKey,
  activeRowIndex,
  activeColumnIndex,
});

export const updateValue = ({ stateKey, value }) => ({
  type: types.CELL_EDITABLE_UPDATE_VALUE,
  stateKey,
  value,
});
