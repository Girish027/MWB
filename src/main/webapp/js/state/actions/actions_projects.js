import fetch from 'isomorphic-fetch';
import _ from 'lodash';
import errorMessageUtil from 'utils/ErrorMessageUtil';
import store from 'state/configureStore';
import * as apiUtils from 'utils/apiUtils';
import getUrl, { pathKey } from 'utils/apiUrls';
import api from 'utils/api';
import { getLanguage } from 'state/constants/getLanguage';
import { RouteNames } from 'utils/routeHelpers';
import Constants from 'constants/Constants';
import Model from 'model';
import * as appActions from './actions_app';
import * as types from './types';

const lang = getLanguage();
const { DISPLAY_MESSAGES } = lang;
// PROJECT

export const requestProject = projectId => ({
  type: types.REQUEST_PROJECT,
  projectId,
});

export const receiveProject = (project, datasets) => ({
  type: types.RECEIVE_PROJECT,
  project,
  datasets,
  receivedAt: Date.now(),
});
// PROJECT LIST

export const toggleSidebarVisible = visible => ({
  type: types.PROJECT_LIST_TOGGLE_VISIBLE,
  visible,
});

export const requestAllProjects = () => ({
  type: types.REQUEST_PROJECTS_ALL,
});

export const receiveAllProjects = projectList => ({
  type: types.RECEIVE_PROJECTS_ALL,
  projectList,
  receivedAt: Date.now(),
});

export const requestProjectsByClient = ({ clientId, startIndex, limit }) => ({
  type: types.REQUEST_PROJECTS_BY_CLIENT,
  clientId,
  startIndex,
  limit,
});

export const receiveProjectsByClient = ({
  projects, clientId, startIndex, limit,
}) => ({
  type: types.RECEIVE_PROJECTS_BY_CLIENT,
  projects,
  clientId,
  startIndex,
  limit,
});

export const sidebarSelectProject = ({ projectId }) => ({
  type: types.SIDEBAR_SELECT_PROJECT,
  projectId,
});

export const collapsibleSidebarSelect = isCollapsibleModelOpen => ({
  type: types.COLLAPSIBLE_SIDEBAR_SELECT,
  isCollapsibleModelOpen,
});

export const projectListLoadedSuccess = newProjectList => ({
  type: types.PROJECT_LIST_LOADED_SUCCESS,
  newProjectList,
});

/* TODO: remove */
export const addDatasetToProjectSuccess = (serverMessage, dataset, projectId) => ({
  type: types.ADD_DATASET_TO_PROJECT_SUCCESS,
  serverMessage,
  dataset,
  projectId,
});

/*
 export const requestDatasetsTransfromationStatuses = (datasetsIds) => {
 return {
 type: types.REQUEST_DATASETS_TRANSFORMATION_STATUSES,
 datasetsIds: datasetsIds
 }
 };
 */
/*
 export const receiveDatasetsTransfromationStatuses = (response) => {
 return {
 type: types.RECEIVE_DATASETS_TRANSFORMATION_STATUSES,
 response: response
 }
 };
 */

export const addNewProject = project => ({
  type: types.ADD_NEW_PROJECT,
  project,
});

export const updateProject = projects => ({
  type: types.PROJECT_UPDATE,
  projects,
});

export const updateProjectStatus = (projectId, status) => ({
  type: types.PROJECT_STATUS_UPDATE,
  projectId,
  status,
});

export const deleteProject = id => ({
  type: types.PROJECT_DELETE,
  id,
});

export const onUpdateProjectSuccess = (response = {}, projectId, message = DISPLAY_MESSAGES.projectUpdated) => (dispatch, getState) => {
  const projects = apiUtils.normalizeIds(response);
  dispatch(updateProject([projects]));
  dispatch(updateProjectStatus(projectId, Constants.STATUS.COMPLETED));
  dispatch(appActions.displayGoodRequestMessage(message));
};

export const onPromoteProjectSuccess = (response = [], projectId) => (dispatch, getState) => {
  const projects = response;
  dispatch(updateProject(projects));
  dispatch(updateProjectStatus(projectId, Constants.STATUS.COMPLETED));
  dispatch(appActions.displayGoodRequestMessage(DISPLAY_MESSAGES.projectPromoted));
};

export const onProjectFailure = (error = {}, projectId) => (dispatch, getState) => {
  const { message } = error;
  dispatch(updateProjectStatus(projectId, Constants.STATUS.FAILED));
  if (error.code === 409) {
    dispatch(appActions.displayWarningRequestMessage(message));
  } else {
    dispatch(appActions.displayBadRequestMessage(message));
  }
};

export const onDeleteProjectSuccess = (projectId) => (dispatch, getState) => {
  const deletedProject = Model.ProjectsManager.getProject(projectId) || {};
  const { name = '' } = deletedProject;
  dispatch(deleteProject(projectId));
  dispatch(appActions.displayGoodRequestMessage(DISPLAY_MESSAGES.projectDeleted(name)));
};

export const onCreateProjectSuccess = (response = {}, clientId, history) => (dispatch, getState) => {
  const project = apiUtils.normalizeIds(response);
  const { name, id: projectId } = project;
  dispatch(addNewProject(project));
  dispatch(appActions.changeRoute(RouteNames.READPROJECT, { clientId, projectId, routeTag: 'datasets' }, history));
  dispatch(appActions.modalDialogChange({
    type: Constants.DIALOGS.DATASET_DIALOG,
    header: Constants.UPLOAD_DATASET,
    dispatch,
    projectId,
    history,
    project,
  }));
  dispatch(appActions.displayGoodRequestMessage(DISPLAY_MESSAGES.projectCreated(name)));
};

export const onCreateProjectFailure = (error = {}) => (dispatch, getState) => {
  const { message } = error;
  dispatch(appActions.modalDialogChange(null));
  if (error.code === 409) {
    dispatch(appActions.displayWarningRequestMessage(message));
  } else {
    dispatch(appActions.displayBadRequestMessage(message));
  }
};

export const updateByProject = data => (dispatch, getState) => {
  const {
    name, description, locale, vertical,
  } = data.values;
  const { clientId, projectId } = data;
  const oldName = getState().projectListSidebar.projectById[projectId].name;
  let formData = [{
    op: Constants.OP.REPLACE,
    path: '/description',
    value: description || '',
  },
  {
    op: Constants.OP.REPLACE,
    path: '/locale',
    value: locale || '',
  },
  {
    op: Constants.OP.REPLACE,
    path: '/vertical',
    value: vertical || 'NULL',
  }];

  if (name !== oldName) {
    formData.push({
      op: Constants.OP.REPLACE,
      path: '/name',
      value: name || '',
    });
  }
  const url = getUrl(pathKey.projectById, { clientId, projectId });
  return new Promise((resolve, reject) => {
    api.patch({
      dispatch,
      getState,
      url,
      headers: {
        'Content-Type': 'application/json-patch+json',
      },
      data: formData,
      onApiSuccess: response => onUpdateProjectSuccess(response, projectId),
      onApiError: error => onProjectFailure(error, projectId),
    });
  });
};

export const deleteProjectById = projectId => (dispatch, getState) => {
  const state = getState();
  const clientId = state.header.client.id;
  const url = getUrl(pathKey.projectById, { projectId, clientId });

  return new Promise((resolve, reject) => {
    api.delete({
      dispatch,
      getState,
      url,
      onApiSuccess: () => onDeleteProjectSuccess(projectId),
      onApiError: error => onProjectFailure(error, projectId),
    });
  });
};

export const promoteProjectById = data => (dispatch, getState) => {
  const state = getState();
  const clientId = state.header.client.id;
  const { projectId } = data;
  const url = getUrl(pathKey.promoteProject, { clientId, ...data });

  return new Promise((resolve, reject) => {
    api.put({
      dispatch,
      getState,
      url,
      onApiSuccess: response => onPromoteProjectSuccess(response, projectId),
      onApiError: error => onProjectFailure(error, projectId),
    });
  });
};

export const demoteProjectById = projectId => (dispatch, getState) => {
  const state = getState();
  const clientId = state.header.client.id;
  const url = getUrl(pathKey.demoteProject, { projectId, clientId });

  return new Promise((resolve, reject) => {
    api.put({
      dispatch,
      getState,
      url,
      onApiSuccess: response => onUpdateProjectSuccess(response, projectId, DISPLAY_MESSAGES.projectDemoted),
      onApiError: error => onProjectFailure(error, projectId),
    });
  });
};

export const createProject = data => (dispatch, getState) => {
  const state = getState();
  const { clientId, history, values } = data;
  const { userId } = state.app;
  const formData = {
    ...values,
    ownerId: userId,
  };

  const url = getUrl(pathKey.projects, { clientId });
  return new Promise((resolve, reject) => {
    api.post({
      dispatch,
      getState,
      url,
      data: formData,
      onApiSuccess: response => onCreateProjectSuccess(response, clientId, history),
      onApiError: onCreateProjectFailure,
    });
  });
};

// Fetch Project & Dataset

const addDatasets = function (project, dispatch) {
  const state = store.getState();
  const { csrfToken } = state.app;
  const clientId = state.header.client.id;

  const fetchUrl = getUrl(pathKey.projectDatasets, { projectId: project.id, clientId });

  return fetch(fetchUrl, {
    credentials: 'same-origin',
    headers: {
      'X-CSRF-TOKEN': csrfToken,

    },
  })
    .then(errorMessageUtil.handleErrors)
    .then(response => response.json())
    .then((datasets) => {
      dispatch(receiveProject(project, datasets));
    })
    .catch(error => errorMessageUtil.dispatchError(error, dispatch, appActions.displayBadRequestMessage));
};

export const fetchProject = projectId => (dispatch, getState) => {
  dispatch(requestProject(projectId));
  const state = getState();
  const { csrfToken } = state.app;
  const clientId = state.header.client.id;

  const fetchUrl = getUrl(pathKey.projectById, { projectId, clientId });

  return fetch(fetchUrl, {
    credentials: 'same-origin',
    headers: {
      'X-CSRF-TOKEN': csrfToken,

    },
  })
    .then(errorMessageUtil.handleErrors)
    .then(response => response.json())
    .then((json) => {
      addDatasets(json, dispatch);
    })
    .catch(error => errorMessageUtil.dispatchError(error, dispatch, appActions.displayBadRequestMessage));
};

export const refreshProjectsByClient = clientId => (dispatch, getState) => {
  // TODO: revert limit back to 100 after fixing MD-153
  dispatch(fetchProjectsByClient({ clientId, startIndex: 0, limit: 500 }));
};

export const fetchProjectsByClient = ({ clientId, startIndex, limit }) => (dispatch) => {
  if (_.isNil(clientId) || clientId == '0') {
    return;
  }
  dispatch(requestProjectsByClient({ clientId, startIndex, limit }));
  const fetchUrl = getUrl(pathKey.projectsForClient, { clientId, startIndex, limit });
  const state = store.getState();
  const { csrfToken } = state.app;
  return fetch(fetchUrl, {
    credentials: 'same-origin',
    headers: {
      'X-CSRF-TOKEN': csrfToken,
    },
  })
    .then(errorMessageUtil.handleErrors)
    .then(response => response.json())
    .then((json) => {
      dispatch(receiveProjectsByClient({
        projects: json, clientId, startIndex, limit,
      }));
    })
    .catch(error => errorMessageUtil.dispatchError(error, dispatch, appActions.displayBadRequestMessage));
};

export const fetchAllProjects = () => (dispatch, getState) => {
  dispatch(requestAllProjects());
  const state = getState();
  const { csrfToken } = state.app;
  const clientId = state.header.client.id;
  const fetchUrl = getUrl(pathKey.projects, { clientId });
  return fetch(fetchUrl, {
    credentials: 'same-origin',
    headers: {
      'X-CSRF-TOKEN': csrfToken,
    },
  })
    .then(errorMessageUtil.handleErrors)
    .then(response => response.json())
    .then((json) => {
      dispatch(receiveAllProjects(json));
    })
    .catch(error => errorMessageUtil.dispatchError(error, dispatch, appActions.displayBadRequestMessage));
};

export const markDeployableModel = ({
  clientId, projectId, modelId, model,
}) => (dispatch, getState) => new Promise((resolve, reject) => {
  const deployableModelId = parseInt(modelId, 10);

  let formData = [{
    op: Constants.OP.REPLACE,
    path: '/deployableModelId',
    value: deployableModelId || null,
  }];

  const url = getUrl(pathKey.projectById, { clientId, projectId });
  return new Promise((resolve, reject) => {
    api.patch({
      dispatch,
      getState,
      url,
      headers: {
        'Content-Type': 'application/json-patch+json',
      },
      data: formData,
      onApiSuccess: response => onUpdateProjectSuccess(response, projectId, DISPLAY_MESSAGES.markModelFlagSuccess(model.version)),
      onApiError: error => onMarkDeployableModelFailure(error, model.version),
    });
  });
});

export const onMarkDeployableModelFailure = (error = {}, modelVersion) => (dispatch) => {
  dispatch(appActions.displayBadRequestMessage(DISPLAY_MESSAGES.markModelFlagFailed(modelVersion)));
};

export const unmarkDeployableModel = ({
  clientId, projectId, model,
}) => (dispatch, getState) => new Promise((resolve, reject) => {
  let formData = [{
    op: Constants.OP.REPLACE,
    path: '/deployableModelId',
    value: null,
  }];

  const url = getUrl(pathKey.projectById, { clientId, projectId });
  return new Promise((resolve, reject) => {
    api.patch({
      dispatch,
      getState,
      url,
      headers: {
        'Content-Type': 'application/json-patch+json',
      },
      data: formData,
      onApiSuccess: response => onUpdateProjectSuccess(response, projectId, DISPLAY_MESSAGES.unmarkModelFlagSuccess(model.version)),
      onApiError: error => onUnmarkDeployableModelFailure(error, model.version),
    });
  });
});

export const onUnmarkDeployableModelFailure = (error = {}, modelVersion) => (dispatch) => {
  dispatch(appActions.displayBadRequestMessage(DISPLAY_MESSAGES.unmarkModelFlagFailed(modelVersion)));
};
