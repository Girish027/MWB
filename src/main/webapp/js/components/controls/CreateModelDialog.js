import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Dialog, Button } from '@tfs/ui-components';
import * as projectActions from 'state/actions/actions_projects';
import Constants from '../../constants/Constants';
import Form from '../Form/Form';
import * as appActions from '../../state/actions/actions_app';
import { pathKey } from '../../utils/apiUrls';
import createModelUiSchema from '../schema/createModel/uiSchema.json';
import createModelJsonSchema from '../schema/createModel/jsonSchema.json';
import CustomTextField from '../fields/CustomTextField';
import CustomDropDown from '../fields/CustomDropDown';

export default class CreateModelDialog extends Component {
  constructor(props) {
    super(props);
    this.form = React.createRef();
    this.props = props;

    this.onClickNext = this.onClickNext.bind(this);
    this.onClickCancel = this.onClickCancel.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    this.onFormChange = this.onFormChange.bind(this);

    this.styleOverride = Object.assign({}, {
      childContainer: {
        marginTop: '0px',
        marginBottom: '0px',
      },
      container: {
        width: '446px',
        height: '460px',
        display: 'grid',
        gridTemplateRows: '60px auto 60px',
      },
    },
    this.props.styleOverride);

    this.state = {
      formData: {},
      isValid: false,
    };

    this.fields = {
      text: CustomTextField,
      locale: (props, url = pathKey.locales) => <CustomDropDown {...props} url={pathKey.locales} />,
      vertical: (props, url = pathKey.verticals) => <CustomDropDown {...props} url={pathKey.verticals} />,
    };
  }

  onClickNext() {
    this.form.current.submit();
  }

  onClickCancel() {
    const { dispatch } = this.props;
    dispatch(appActions.modalDialogChange(null));
  }

  onSubmit() {
    const { formData } = this.state;
    const { dispatch, clientId, history } = this.props;
    let data = {
      clientId,
      history,
      values: { clientId, ...formData },
    };
    dispatch(projectActions.createProject(data));
  }

  onFormChange(data, errors) {
    if (errors.length === 0) {
      this.setState({
        formData: data,
        isValid: true,
      });
    } else if (errors.length > 0) {
      this.setState({
        isValid: false,
      });
    }
  }

  render() {
    const {
      header, size, formData, ...otherProps
    } = this.props;

    const { isValid } = this.state;

    return (
      <div>
        <Dialog
          size={size}
          headerChildren={header}
          {...otherProps}
          isOpen
          okDisabled={!isValid}
          showBodySeperator={false}
          okVisible
          okChildren={Constants.NEXT}
          cancelVisible
          cancelChildren={Constants.CANCEL}
          closeIconVisible
          styleOverride={this.styleOverride}
          onClickOk={this.onClickNext}
          onClickCancel={this.onClickCancel}
          onClickClose={this.onClickCancel}
          centerContent={false}
        >
          <div>
            <div>
              <Form
                uiSchema={createModelUiSchema}
                jsonSchema={createModelJsonSchema}
                fields={this.fields}
                onSubmit={this.onSubmit}
                onChange={this.onFormChange}
                formData={formData}
                ref={this.form}
                liveValidate
              />
            </div>
          </div>
        </Dialog>
      </div>
    );
  }
}

CreateModelDialog.propTypes = {
  dispatch: PropTypes.func,
  history: PropTypes.object,
  header: PropTypes.string,
  closeIconVisible: PropTypes.bool,
  onClickOk: PropTypes.func,
  onClickClose: PropTypes.func,
  styleOverride: PropTypes.object,
  size: PropTypes.string,
  formData: PropTypes.object,
  errorType: PropTypes.string,
};

CreateModelDialog.defaultProps = {
  dispatch: () => {},
  history: {},
  closeIconVisible: true,
  onClickOk: () => {},
  onClickClose: () => {},
  styleOverride: {},
  size: 'large',
  formData: {},
  errorType: '',
  header: Constants.CREATE_MODEL,
};
