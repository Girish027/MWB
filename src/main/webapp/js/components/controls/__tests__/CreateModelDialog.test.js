import React from 'react';
import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import * as actionsApp from 'state/actions/actions_app';
import Constants from 'constants/Constants';
import * as projectActions from 'state/actions/actions_projects';
import CreateModelDialog from 'components/controls/CreateModelDialog';

describe('<CreateModelDialog />', () => {
  let wrapper;
  let mockPush = jest.fn();
  let history = {};
  const props = {
    header: Constants.CREATE_MODEL,
    onCancel: () => {},
    dispatch: () => {},
    history,
    clientId: '14',
  };

  const getShallowWrapper = (propsObj) => shallow(<CreateModelDialog
    {...props}
  />);

  afterAll(() => {
    jest.clearAllMocks();
  });

  describe('Creating an instance with test props', () => {
    beforeEach(() => {
      wrapper = getShallowWrapper(props);
    });

    test('should exist', () => {
      expect(wrapper.exists()).toBe(true);
    });

    test('Snapshots should render correctly', () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });

  describe('Function Call Handler:', () => {
    beforeAll(() => {
      props.header = Constants.CREATE_MODEL,
      props.dispatch = jest.fn();
      props.onCancel = jest.fn();
      actionsApp.modalDialogChange = jest.fn(() => 'called modalDialogChange');
      projectActions.createProject = jest.fn(() => 'createProject');
    });

    beforeEach(() => {
      wrapper = shallow(<CreateModelDialog
        {...props}
      />);
    });

    afterEach(() => {
      jest.clearAllMocks();
    });

    describe('onClickCancel:', () => {
      test('should onClickCancel call click handler', () => {
        wrapper.instance().onClickCancel();
        expect(props.dispatch).toHaveBeenCalledWith('called modalDialogChange');
        expect(actionsApp.modalDialogChange).toHaveBeenCalledWith(null);
      });
    });

    describe('onSubmit:', () => {
      test('should call create project', () => {
        const { dispatch, clientId, history } = props;
        const formData = {
          name: 'testProject',
          description: 'testProject',
          vertical: 'FINANCIAL',
          locale: 'en-US',
        };
        wrapper.setState({
          formData,
        });
        wrapper.instance().onSubmit();
        expect(projectActions.createProject).toHaveBeenCalledWith({
          clientId,
          history,
          values: { clientId, ...formData },
        });
        expect(dispatch).toHaveBeenCalledWith('createProject');
      });
    });

    describe('onFormChange:', () => {
      test('should change the state when there are no error', () => {
        const data = {
          name: 'abc',
          description: 'form',
        };
        wrapper.instance().onFormChange(data, []);
        expect(wrapper.state().formData).toBe(data);
        expect(wrapper.state().isValid).toBe(true);
      });

      test('should change the state when there are no error', () => {
        const data = {
          name: 'abc',
          description: 'form',
        };
        wrapper.instance().onFormChange(data, [{ name: 'is required' }]);
        expect(wrapper.state().isValid).toBe(false);
      });
    });
  });
});
