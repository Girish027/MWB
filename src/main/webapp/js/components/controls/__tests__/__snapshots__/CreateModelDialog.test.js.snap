// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateModelDialog /> Creating an instance with test props Snapshots should render correctly 1`] = `
<div>
  <withTheme(DialogComponent)
    cancelChildren="CANCEL"
    cancelVisible={true}
    centerContent={false}
    clientId="14"
    closeIconVisible={true}
    dispatch={[Function]}
    errorType=""
    headerChildren="Create New Model"
    history={Object {}}
    isOpen={true}
    okChildren="NEXT"
    okDisabled={true}
    okVisible={true}
    onCancel={[Function]}
    onClickCancel={[Function]}
    onClickClose={[Function]}
    onClickOk={[Function]}
    showBodySeperator={false}
    size="large"
    styleOverride={
      Object {
        "childContainer": Object {
          "marginBottom": "0px",
          "marginTop": "0px",
        },
        "container": Object {
          "display": "grid",
          "gridTemplateRows": "60px auto 60px",
          "height": "460px",
          "width": "446px",
        },
      }
    }
  >
    <div>
      <div>
        <Form
          fields={
            Object {
              "locale": [Function],
              "text": [Function],
              "vertical": [Function],
            }
          }
          formContext={Object {}}
          formData={Object {}}
          jsonSchema={
            Object {
              "properties": Object {
                "description": Object {
                  "optional": true,
                  "title": "Description",
                  "type": "string",
                },
                "locale": Object {
                  "default": "en-US",
                  "title": "Locale",
                  "type": "string",
                },
                "name": Object {
                  "isRequired": true,
                  "title": "Name",
                  "type": "string",
                },
                "vertical": Object {
                  "default": "FINANCIAL",
                  "title": "Vertical",
                  "type": "string",
                },
              },
              "required": Array [
                "name",
              ],
              "title": "",
              "type": "object",
            }
          }
          liveValidate={true}
          onChange={[Function]}
          onSubmit={[Function]}
          showErrorList={false}
          uiSchema={
            Object {
              "description": Object {
                "ui:field": "text",
              },
              "locale": Object {
                "ui:field": "locale",
              },
              "name": Object {
                "ui:field": "text",
              },
              "vertical": Object {
                "ui:field": "vertical",
              },
            }
          }
        />
      </div>
    </div>
  </withTheme(DialogComponent)>
</div>
`;
